presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/SemiJoinMatcher.java

```diff
@@ -56,6 +60,10 @@ public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session ses
             return NO_MATCH;
         }
 
+        if (distributionType.isPresent() && !distributionType.equals(semiJoinNode.getDistributionType())) {
```

Comment:
This semantic is confusing. It is not clear that if `distributionType` is not set that means that any `distributionType` is allowed. I would change the semantics to the strict match if possible. So if `distributionType` is empty - check that actual distribution type is empty.

Reply:
This is the same semantic as for JoinMatcher.  The reason for this is to have a way to say "I don't care about the distribution type here".   We don't every use that for SemiJoin, but I'd be hesitant to make semi-join different from join.  I tried making join an exact check, and there are three test failures.  If you think it's better I can fix those 3 tests to check for a specific distribution type and then change this.